pageserver/src/tenant.rs

```diff
@@ -340,18 +342,26 @@ impl Tenant {
 
     /// Get Timeline handle for given Neon timeline ID.
     /// This function is idempotent. It doesn't change internal state in any way.
-    pub fn get_timeline(&self, timeline_id: TimelineId) -> anyhow::Result<Arc<Timeline>> {
-        self.timelines
-            .lock()
-            .unwrap()
-            .get(&timeline_id)
-            .with_context(|| {
-                format!(
-                    "Timeline {} was not found for tenant {}",
-                    timeline_id, self.tenant_id
-                )
-            })
-            .map(Arc::clone)
+    pub fn get_timeline(
+        &self,
+        timeline_id: TimelineId,
+        active_only: bool,
+    ) -> anyhow::Result<Arc<Timeline>> {
```

Comment:
I would prefer a separate `get_active_timeline` function over a boolean argument. When you see a call with a boolean arg, you need to always look at the function definition to see what true and false means.

Or make it an enum with two values, Active / Any. But I think `get_active_timeline` is best. (If you remember, I'm doing the same with `get_active_tenant` in the on-demand download PR)

Reply:
I agree generally, yet would propose to do it in another PR: we need to do the same thing for tenant too, and for that it appears that we could completely hide inactive tenants from public API if we return `TenantInfo` from `Tenant` itself.

I would explore similar capabilities for timeline in that separate PR.